src/ipc/cc_version.c

```diff
@@ -0,0 +1,28 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2016 Intel Corporation. All rights reserved.
+//
+// Author: Karol Trzcinski <karolx.trzcinski@linux.intel.com>
+
+#include <sof/compiler_info.h>
+#include <ipc/info.h>
+
+static const struct sof_ipc_cc_version cc_version = {
+	.ext_hdr	= {
+		.hdr.cmd = SOF_IPC_FW_READY,
+		.hdr.size = sizeof(struct sof_ipc_cc_version)
+			    + sizeof(CC_DESC),
+		.type = SOF_IPC_EXT_CC_INFO,
+	},
+	.micro = CC_MICRO,
+	.minor = CC_MINOR,
+	.major = CC_MAJOR,
+	CC_NAME_COPY(.name),
+	CC_OPTIM_COPY(.optim),
+	.desc = CC_DESC,
+};
+
+const struct sof_ipc_cc_version *get_ipc_cc_version(void)
+{
+	return &cc_version;
+}
```

Comment:
No need for function there if there is generic implementation of this struct. We could use function if we'd expect different platforms to implement it in different ways. Here we can just simplify things and make `cc_version` globally available, but keeping it const as you did (what is good).